node_modules/oae-zencoder/lib/api.js

```diff
@@ -0,0 +1,122 @@
+
+    ZencoderDAO.getContentRevisionId(zencoderId, function(err, contentRevision) {
+        if (err) {
+            return callback(err);
+        }
+
+        return _createAndSavePreviews(contentRevision, output, job['pass_through'], callback);
+    });
+};
+
+/**
+ * Create and save the previews
+ *
+ * @param  {Object}     contentRevision   An object containing content item and revision IDs
+ * @param  {Object}     output            The output information from Zencoder
+ * @param  {String}     key               The path to the video file on Amazon S3
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ * @api private
+ */
+var _createAndSavePreviews = function(contentRevision, output, key, callback) {
+    var contentId = contentRevision.contentId;
+    var ctx = new PreviewContext(PreviewProcessorAPI.getConfiguration(), contentId, contentRevision.revisionId);
+
+    log().info('Received encoding completed notification from Zencoder for contentId %s', contentId);
+
+    ctx.login(function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        ctx.getContentData(function(err) {
+            if (err) {
+                return callback(err);
+            } else if (ctx.revision.previews.total > 0) {
+                return callback({'code': 500, 'msg': 'Previews have already been created for this revision'});
+            }
+
+            ctx.addPreviewMetadata('videoKey', key);
+
+            var thumbnail = _.findWhere(output.thumbnails, {'label': 'base'});
+
+            if (thumbnail || _.isArray(thumbnail.images)) {
+                return _createThumbnails(ctx, _.first(thumbnail.images), callback);
+            } else {
+                // If there are no thumbnail images, we just save the video url
+                return ctx.savePreviews(callback);
+            }
+        });
+    });
+};
+
+
+/**
+ * Create and save thumbnails
+ *
+ * @param  {Object}     ctx               The preview context for this content item
+ * @param  {Object}     thumbnailImage    The thumbnail image data from Zencoder
+ * @param  {Function}   callback          Standard callback function
+ * @param  {Object}     callback.err      An error that occurred, if any
+ * @api private
+ */
+var _createThumbnails = function(ctx, thumbnailImage, callback) {
+    if (!thumbnailImage.url) {
+        return callback();
+    }
+    PreviewUtil.downloadRemoteFile(thumbnailImage.url, util.format('%s/base.png', ctx.baseDir), function(err, path) {
+        if (err) {
+            return callback(err);
+        }
+        PreviewUtil.generatePreviewsFromImage(ctx, path, {'removeInput': true}, function(err) {
+            if (err) {
+                return callback(err);
+            }
+            log().info('Saved preview and thumbnails for video content item %s', ctx.content.contentId);
+            return ctx.savePreviews();
+        });
+    });
+};
```

Comment:
Files need to end w/ a new line. FWIW, there is an `.editorconfig`, that describes small spacing things like this and editors like Sublime and Atom have plugins to manage it for you.